handle the case x>k

fm94 opened the pull request:
Raise an `AssertionError` if the number of NN is larger than then the number of observers

fm94 committed:
handle the case x>k

Raise an `AssertionError` if the number of NN is larger than then the number of observers

alexhartl commented:
I agree that raising an error is the way to go here. But this check will cause problems if `k` is `None`, i.e. when using the pca functionality. So check for `k` being `None` and also add it after line 127?

fm94 committed:
Move the AssertationError

The condition is checked after the final 'k' initialization. Hence considering also the case when 'k' is `None`

fm94 commented:
This is the proper way to do it.

Commit:
use a double check for k being smaller than x

alexhartl commented:
Thanks
